### ISSUE TITLE
Feature request: Add LFS - Large File Support

### ISSUE THREAD

**Issue by myplacedk** [maks/MGit#130]
LFS solves some problems with binary files in Git repositories. Unfortunately, LFS also mean I can't access the binary files on my Android devices, where I use MGit.

I have no idea how much work it takes to add LFS to MGit, or how much larger it will make the APK.

**Comment by maks** [comment]
Agreed, I'd like to support LFS. I'll need to look into what that will involve.

**Comment by piccobit** [comment]
Any news regarding the LFS support?

**Comment by maks** [comment]
Sorry this is a very big piece of new functionality, I don't expect I'll get a chance to work on it this year, but I'm very happy for contributions to implement this.

**Comment by KOLANICH** [comment]
https://github.com/eclipse/jgit/tree/master/org.eclipse.jgit.lfs

**Comment by maks** [comment]
@KOLANICH sorry as I mentioned in https://github.com/maks/MGit/issues/259#issuecomment-368353413 MGit is stuck with using an older version of JGit for now until a solution is available for #26.

**Comment by ShivamMukherjee** [comment]
An implementation of LFS without using JGit (iinw): [LFGit by MarekPetr](https://github.com/MarekPetr/LFGit).

This may also solve #26 unless that has already been dealt with.

**Comment by maks** [comment]
Thank you for the suggestion to look at LFGit but it appears it's using a somewhat convoluted implementation via JNI which I'm not sure I want to commit to at this stage.

**Comment by ShivamMukherjee** [comment]
You don't need to implement it that way, but somebody implemented it and this issue has been up for three years. If it's harder to adapt jGit to allow LFS support, remember that even an obscure app using a convoluted JNI based mechanism is able to do it.

Please implement. Your app is amazing and LFS is fairly popular, especially in the indie games scene.

**Comment by maks** [comment]
@Vivraan I'm happy to look at a possible implementation using LFGit but I think you will find that it will not be a simple case to integrate that in MGit as the canonical git implementation uses smudge filters to implement LFS and that is only available [after JGit 4.6](http://archive.eclipse.org/jgit/site/4.6.1.201703071140-r/apidocs/org/eclipse/jgit/lfs/SmudgeFilter.html)
Otherwise you would need to do the integration of LFS "manually" in MGit, working around the lack of direct support in JGit and in that case you pretty much might as well implement handling the LFS pointer files directly in MGit anyway as [the spec is actually pretty straightforward](https://github.com/git-lfs/git-lfs/blob/master/docs/spec.md).

**Comment by kb-1000** [comment]
So, what's needed to get JGit 4+? Just a java.nio.file implementation, right?

**Comment by maks** [comment]
@kb-1000 as far as I know, but JGit is up to 5.8 release and I really haven't been tracking the changes closely so there may know be other Java SDK features being use that are not supported on Android.

**Comment by Vivraan** [comment]
JGit's dependencies are all on 1.8.

**Comment by aivanovski** [comment]
In order to work with LFS, JGit under the hood executes "git-lfs" binary on the host machine. I see 2 possible solutions for this issue:
1 - replace JGit with something else
2 - provide 'git-lfs' binary with Termux, but I'm not sure that this is possible, because it requires access to a private Termux directory

A bit more information in a comment here https://github.com/maks/MGit/issues/26#issuecomment-927110747

**Comment by lucastucious** [comment]
Any news about LFS support?

**Comment by thegrojas** [comment]
I'm also interested in LFS support
